File: code/classes/pattern-matching.js
// Pattern matching in JavaScript (using switch or if-else chains)
function describe(obj) {
    if (obj === 1) {
        return "One";
    } else if (obj === "Hello") {
        return "Greeting";
    } else if (typeof obj === 'number' && Number.isInteger(obj)) {
        return "Integer";
    } else if (typeof obj === 'string') {
        return "String";
    } else {
        return "Unknown";
    }
}

console.log(describe(1));
console.log(describe("Hello"));
console.log(describe(3.14));

File: code/classes/pattern-matching.go
// Pattern matching in Go (using type switch)
func describe(obj interface{}) string {
    switch v := obj.(type) {
    case int:
        if v == 1 {
            return "One"
        }
        return "Integer"
    case string:
        if v == "Hello" {
            return "Greeting"
        }
        return "String"
    case float64:
        return "Float"
    default:
        return "Unknown"
    }
}

func main() {
    fmt.Println(describe(1))
    fmt.Println(describe("Hello"))
    fmt.Println(describe(3.14))
}
